File: BackEnd/src/domain/exercise-result.ts
import {
  ExerciseResultId,
  StudentId,
  SupportItemId,
  SupportProgramId,
  SupportProgramMilestoneId,
} from './ids';
import { SubmissionStatus } from './quiz-result';

export class ExerciseResult {
  exerciseResultId: ExerciseResultId;
  studentId: StudentId;
  supportProgramId: SupportProgramId;
  milestoneId: SupportProgramMilestoneId;
  supportItemId: SupportItemId;
  passed: boolean;
  status: SubmissionStatus;
  answer?: string;
  feedback?: string;
  submittedAt: Date;
  reviewedAt?: Date;
}

File: Frontend/GP/Models/ExerciseResult.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class ExerciseResult : BaseEntity
    {
        [Key]
        public int ExerciseResultId { get; set; }

        public int StudentId { get; set; }
        public Student Student { get; set; }

        public  int SupportProgramId { get; set; }
        public SupportProgram SupportProgram { get; set; }

        public int MilestoneId { get; set; }
        public Milestone Milestone { get; set; }

        public int SupportItemId { get; set; }
        public SupportItem SupportItem { get; set; }

        public bool Passed { get; set; }

        public string Status { get; set; }

        public string? Answer { get; set; }

        public string? Feedback { get; set; }

        public DateTime SubmittedAt { get; set; }

        public DateTime? ReviewedAt { get; set; }
    }
}
